test/ci/QuickOpenTest.ts

```diff
@@ -1,21 +1,50 @@
-/**
- * Test scripts for QuickOpen
- */
+import * as Oni from "oni-api"
 
+import { Assertor } from "./Assert"
 import { waitForCommand } from "./Common"
 
-export const test = async (oni: any) => {
+export const test = async (oni: Oni.Plugin.Api) => {
+    const assert = new Assertor("QuickOpen")
     await oni.automation.waitForEditors()
+    await oni.automation.waitFor(() => oni.plugins.loaded)
+    const instance = oni.plugins.getPlugin("oni-plugin-quickopen")
+    assert.defined(instance, "plugin instance")
 
-    await waitForCommand("quickOpen.show", oni)
-
-    oni.commands.executeCommand("quickOpen.show")
+    // Start
+    oni.commands.executeCommand("quickOpen.searchFileByPath")
     await oni.automation.waitFor(() => oni.menu.isMenuOpen())
 
-    await oni.automation.sleep(500)
-
     await waitForCommand("menu.close", oni)
-
     oni.commands.executeCommand("menu.close")
     await oni.automation.waitFor(() => !oni.menu.isMenuOpen())
+    // End
+
+    /*
+    // Start
+    await waitForCommand("quickOpen.searchFileByPath", oni)
+    oni.commands.executeCommand("quickOpen.searchFileByPath")
+    await oni.automation.waitFor(() => oni.menu.isMenuOpen())
+
+    const anyOni = oni as any
+    await oni.automation.sleep(800)
+    anyOni.automation.sendKeysV2("searchprovider.ts")
+    await oni.automation.sleep(100)
+    anyOni.automation.sendKeysV2("<CR>")
+    const editor = oni.editors.activeEditor
+    await oni.automation.waitFor(() => editor.activeBuffer.filePath.endsWith("/SearchProvider.ts"))
+    // End
+
+    // Start
+    await waitForCommand("quickOpen.searchFileByContent", oni)
+    oni.commands.executeCommand("quickOpen.searchFileByContent")
+    await oni.automation.waitFor(() => oni.menu.isMenuOpen())
+    await oni.automation.sleep(100)
+    anyOni.automation.sendKeysV2("^exp.*?class.*?Qu.*?Op.*?Item")
+    await oni.automation.sleep(100)
+    anyOni.automation.sendKeysV2("<CR>")
+    await oni.automation.sleep(500)
+    const filePath = oni.editors.activeEditor.activeBuffer.filePath
+    assert.assert(filePath.endsWith("QuickOpenItem.ts"), `filePath`)
```

Comment:
I was hitting issues with this assert where the test was finding itself! So it would search for some string and the first occurrence would be the test file itself. I think if  you swap the search string to something unique to the test (`MyUniqueTestString`) and then assert the file is the test file, you should get better results.